### PR TITLE
standard:5.0 Golang version 1.18.1 support and bump to 1.16.15, 1.17.8

### PR DESCRIPTION
**Pull request by friimaind**
Closes Issues  [#518](https://github.com/aws/aws-codebuild-docker-images/issues/518) and [#478](https://github.com/aws/aws-codebuild-docker-images/pull/478)

*Description of changes:*
```aws/codebuild/standard:5.0``` image changes

- Add support for ```golang: 1.18.1```
- Bump go 1.15 to latest version ```golang: 1.15.15```
- Bump go 1.16 to latest version ```golang: 1.16.15```
- Bump go 1.17 to latest version ```golang: 1.17.8```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by louie-pulse-id — APPROVED** [review]

**Comment by joekendal** [comment]
+1

**Comment by joekendal** [comment]
@clareliguori @leoherran-aws please 🙏😭

**Review by louie-pulse-id — APPROVED** [review]
we all need this

**Review by albertoforcato — APPROVED** [review]

**Review by lechgajewski — APPROVED** [review]

**Review by SnaKKo — APPROVED** [review]

**Comment by mirpo** [comment]
+1

**Review by mirpo — APPROVED** [review]
LGTM

**Review by LucaBortolettoHevelop — APPROVED** [review]

**Comment by joekendal** [comment]
temporary workaround for anyone stuck on this. add the following to your script:

```
cd $HOME/.goenv && git pull --ff-only && cd -
goenv install 1.17.0
goenv local 1.17.0
```

**Comment by mirpo** [comment]
@joekendal for some reason it didn't work for me: `goenv: command not found`
but I have working solution to share.

```
version: 0.2

phases:
  install:
    commands:
      ###### TEMPORARY WORKAROUND to install 1.18
      # remove after this merged https://github.com/aws/aws-codebuild-docker-images/pull/519
      - wget -q https://storage.googleapis.com/golang/go1.18.linux-amd64.tar.gz
      - tar -C /usr/local -xzf go1.18.linux-amd64.tar.gz
      - export PATH="/usr/local/go/bin:$PATH" && export GOPATH="$HOME/go" && export PATH="$GOPATH/bin:$PATH"
      - go version
      ######
```

result:
```


[Container] 2022/04/14 07:03:11 Running command wget -q https://storage.googleapis.com/golang/go1.18.linux-amd64.tar.gz
18 |  
19 | [Container] 2022/04/14 07:03:13 Running command tar -C /usr/local -xzf go1.18.linux-amd64.tar.gz
20 |  
21 | [Container] 2022/04/14 07:03:16 Running command export PATH="/usr/local/go/bin:$PATH" && export GOPATH="$HOME/go" && export PATH="$GOPATH/bin:$PATH"
22 |  
23 | [Container] 2022/04/14 07:03:16 Running command go version
24 | go version go1.18 linux/amd64
```

**Comment by lechgajewski** [comment]
I'm really sorry to write this and my aim is not to force merge the PR with force... but `runtimes.yml` was updated 9 months ago...? At the modern pace/schedules being released and calling this the AWS Standard CodeBuild Docker Images... well this is pretty bad 🙁  Since the time this PR was opened, Go went from 1.18 to 1.18.1 already.

I really like using these Images, so 🤞  there is something official soon and we don't have to use workarounds to get things done - 🙏  AWS...

**Review by Omnia89 — APPROVED** [review]

**Comment by joekendal** [comment]
> @joekendal for some reason it didn't work for me: `goenv: command not found` but I have working solution to share.

Can you confirm you have this error on aws/codebuild/standard:5.0 ?

**Comment by joekendal** [comment]
> At the modern pace/schedules of software being released - calling this the AWS Standard CodeBuild Docker Images... well this is pretty bad 🙁 

Exactly, this is breaking everyone's pipelines

File: ubuntu/standard/5.0/Dockerfile [hunk 1]
```diff
@@ -298,11 +298,15 @@ RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/bin -
...
-    goenv install $GOLANG_16_VERSION && rm -rf /tmp/* && \
+ENV GOLANG_15_VERSION="1.15.15"
+ENV GOLANG_16_VERSION="1.16.15"
+ENV GOLANG_17_VERSION="1.17.8"
+ENV GOLANG_18_VERSION="1.18"
+
+RUN goenv install $GOLANG_15_VERSION; rm -rf /tmp/*; \
+    goenv install $GOLANG_16_VERSION; rm -rf /tmp/*; \
+    goenv install $GOLANG_17_VERSION; rm -rf /tmp/*; \
+    goenv install $GOLANG_18_VERSION; rm -rf /tmp/*; \
```
Comment (hoffa) [inline ubuntu/standard/5.0/Dockerfile:309]: Why the change from `&&` to `;`? Are the `goenv` commands expected to fail?

Comment (hoffa) [inline ubuntu/standard/5.0/Dockerfile:309]:
```suggestion
RUN goenv install $GOLANG_15_VERSION && rm -rf /tmp/* && \
    goenv install $GOLANG_16_VERSION && rm -rf /tmp/* && \
    goenv install $GOLANG_17_VERSION && rm -rf /tmp/* && \
    goenv install $GOLANG_18_VERSION && rm -rf /tmp/* && \
```

**Review by albertoforcato — APPROVED** [review]

File: ubuntu/standard/5.0/Dockerfile [hunk 2]
```diff
@@ -298,11 +298,15 @@ RUN curl -sS https://getcomposer.org/installer | php -- --install-dir=/usr/bin -
 #****************      END PHP     ****************************************************
 
 #****************     GOLANG     ****************************************************
-ENV GOLANG_15_VERSION="1.15.12"
-ENV GOLANG_16_VERSION="1.16.4"
+ENV GOLANG_15_VERSION="1.15.15"
+ENV GOLANG_16_VERSION="1.16.15"
+ENV GOLANG_17_VERSION="1.17.8"
+ENV GOLANG_18_VERSION="1.18"
```
Comment (lechgajewski) [inline ubuntu/standard/5.0/Dockerfile:304]: There is a PR for `goenv` supporting `1.18.1` opened already:

```suggestion
ENV GOLANG_18_VERSION="1.18.1"
```

Since this PR has been opened for a longer while now, maybe it's not much of a timing gamble to suggest this?

Comment (lechgajewski) [reply ubuntu/standard/5.0/Dockerfile]: @friimaind Glad that you like the idea - could you update the PR description as well (just for clarity)?

`Add support for golang: 1.18.1`

**Comment by friimaind** [comment]
@lechgajewski sure! PR comment updated

**Review by albertoforcato — APPROVED** [review]

**Review by albertoforcato — APPROVED** [review]

**Review by helmus — APPROVED** [review]
Would love to see support of 1.18.1 🙏

**Review by kelongchen — APPROVED** [review]

**Comment by givsly-stephen** [comment]
+1

**Comment by johannes-gehrs** [comment]
I am begging the maintainers of this repo to finally update the Go versions...

**Comment by mnugter** [comment]
Please update the allowed Go versions as bitbucket has now twice broken our code pipelines due to problems with Golang versions.

At least upgrade 1.16 to the latest version so we can use bitbucket repo's again.

**Comment by friimaind** [comment]
Bitbucket just broke our pipelines **again**, please can we have an update for this PR?

**Review by frenkyoptic — APPROVED** [review]

**Comment by subinataws** [comment]
Latest version of Go added in https://github.com/aws/aws-codebuild-docker-images/releases/tag/22.06.30
